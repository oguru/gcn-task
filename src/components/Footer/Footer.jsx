import { Link } from "@reach/router";
import React, { useEffect, useRef } from "react";
import SocialMediaIcon from "../SocialMediaIcon";
import gcnLogo from "../../assets/GCNLogo.svg";
import styles from "./Footer.module.scss";

const Footer = (props) => {
const {setFooterHeight} = props;

const inputRef = useRef(null);

useEffect(() => {
  const height = inputRef.current.offsetHeight;
  setFooterHeight(height);
}, [inputRef, setFooterHeight]); //error

  return (
    <>
      <footer 
        className={styles.footer}
        ref={inputRef} 
      >
        <div className={styles.footerTop}>
          <Link to={"/"}>
            <img 
              alt="GCN Logo" 
              className={styles.gcnLogo} 
              src={gcnLogo}
            />
          </Link>
          <div className={styles.socialMediaIcons}>
            <SocialMediaIcon
              isPrimary={false} 
              icon="facebook"
              key="facebookf"
              link="https://www.facebook.com/globalcyclingnetwork" 
            />
            <SocialMediaIcon 
              isPrimary={false} 
              icon="twitter" 
              key="twitterf"
              link="https://twitter.com/gcntweet" 
            />
            <SocialMediaIcon 
              isPrimary={false} 
              icon="instagram" 
              key="instagramf"
              link="https://www.instagram.com/globalcyclingnetwork/?hl=en"
            />
          </div>
        </div>
        <div className={styles.footerBot}>
          <div className={styles.footerLinks}>
            <div className={styles.footerCol}>
              <p>Editorial Policy</p>
              <p>Privacy Policy</p>
              <p>Cookie Policy</p>
            </div>
            <div className={styles.footerCol}>
              <p>Promotional Rules</p>
              <p>Terms &amp; Conditions</p>
              <p>Youtube Terms &amp; Conditions</p>
            </div>
          </div>          
        </div>
        <div className={styles.footerLinks}>
          <p>&copy; 2020 Play Sports Network </p>
          <p>info@playsportsnetwork.com</p>
        </div>
      </footer>
    </>
  );
};

export default Footer;
