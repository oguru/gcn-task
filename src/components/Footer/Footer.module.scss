@import "../../variables.scss";

.footer {
    background-color: black;
    bottom: 0;
    color: white;
    padding: 0 $base-spacing*2;
    position: absolute;
    width: 100%;

    p {
        font-size: 12px;
        line-height: 25px;
    }

    .bottomText {
        display: flex;
    }

    .footerLinks {
        border-top: solid 1px rgba(255, 255, 255, 1);
        display: flex;
        padding: $base-spacing 0;
        justify-content: space-between;
    }

    .footerTop {
        align-items: center;
        display: flex;
        justify-content: space-between;
        padding: $base-spacing 0;

        .gcnLogo {
            height: 40px;
            width: 100px;
        }

        .socialMediaIcons {
            display: flex;
        }
    }
}