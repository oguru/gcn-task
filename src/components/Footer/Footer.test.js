import React from "react";
import { mount } from "enzyme";
import { render, cleanup } from "@testing-library/react";
import Footer from "./Footer";

describe("Footer tests", () => {
  let component;
  const setFooterHeight = jest.fn();

  afterEach(cleanup);

  it("should render", () => {
    expect(render(<Footer setFooterHeight={setFooterHeight}/>)).toBeTruthy();
  });

  beforeEach(() => {
    component = mount(<Footer setFooterHeight={setFooterHeight}/>);
  });

  it("should match the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should contain a footer tag", () => {
    expect(component.find("footer").length).toEqual(1);
  });

  it("should have 3 social media icons", () => {
    expect(component.find("SocialMediaIcon").length).toBe(3);
  })
});
