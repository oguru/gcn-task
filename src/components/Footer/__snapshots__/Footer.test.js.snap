// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests should match the snapshot 1`] = `ReactWrapper {}`;
